application_form/api/views.py

```diff
@@ -1,22 +1,49 @@
-from rest_framework.permissions import IsAuthenticated
+from django.views.decorators.http import require_http_methods
+from rest_framework import permissions, status
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.exceptions import NotFound, ValidationError
+from rest_framework.response import Response
 
+from apartment.models import Project
 from application_form.api.serializers import (
     ApplicationSerializer,
     SalesApplicationSerializer,
 )
+from application_form.exceptions import ProjectDoesNotHaveApplicationsException
 from application_form.models import Application
+from application_form.services.lottery.machine import distribute_apartments
 from audit_log.viewsets import AuditLoggingModelViewSet
 from users.permissions import IsSalesperson
 
 
+@api_view(http_method_names=["POST"])
+@permission_classes([permissions.AllowAny])
+@require_http_methods(["POST"])  # For SonarCloud
+def execute_lottery_for_project(request):
```

Comment:
I would recommend using a serializer here since that would take care of validation of the UUID for you and also handled the error messages.

Reply:
Fixed